### ISSUE TITLE
Events with long description can not be created or modified ("String data, right truncated: 1406 Data too long for column 'value' at row 1")

### ISSUE THREAD

**Issue by bcutter** [nextcloud/server#16467]
### Steps to reproduce
1. Create a calendar event with a very long description (notes) or try to modify an existing one - doesn´t matter on which endpoint or even on the NC calendar webinterface itself
2. Try to sync the calendar
3. See nextcloud error log

### Expected behaviour
calendar event is synced

### Actual behaviour
error log entry is created, over time calendars of different endpoints get completely out of sync (no error message visible to the user!)

### Server configuration

**Operating system**: Raspbian 10 Buster (hardware: Raspberry Pi 3 B+)

**Web server:** nginx 1.14.2

**Database:** MariaDB 10.3.15, InnoDB database structure

**PHP version:** PHP 7.3.4-2

**Server version:** 16.0.3

**Calendar version:** 1.7.0

**Updated from an older installed version or fresh install:** updated

#### Log file (data/nextcloud.log)
```
Fatal	webdav	Doctrine\DBAL\Exception\DriverException: An exception occurred while executing 'INSERT INTO `nc_calendarobjects_props` (`calendarid`, `calendartype`, `objectid`, `name`, `parameter`, `value`) VALUES(?, ?, ?, ?, ?, ?)' with params ["4", 0, 7854, "DESCRIPTION", null, "Very long description text here\n\n Very long description text here\n\n Very long description text here\n\n Very long description text here\n\n Very long description text here\n\n Very long description text here\n\n Very long description text here\n\n Very long description te"]: SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'value' at row 1
```

==> What to do, what to check, how to help you help me?

**Comment by bcutter** [comment]
I checked column definition for column "value" in table "nc_calendarobjects_props": varchar(191)

Changed it by alter table to "varchar(767)" which is the highest possible value, triggered sync of affected calendar events with quite/very long description and: no error log entry, calendard event is being synced as expected. So this seems like a quick and dirty workaround.

What is the default column size definition for that "value" column?

**Comment by kesselb** [comment]
https://github.com/nextcloud/server/blob/4a5282ba2191059a356540165ac43dcb972eda8d/apps/dav/lib/Migration/Version1004Date20170825134824.php#L438-L441

> What is the default column size definition for that "value" column?

255

**Comment by bcutter** [comment]
Changed it back to the default (255) and didn´t see this error/issue again so far. I have no idea why and when (Nextcloud upgrade? To v16 or even before on v15?) this number changed.

Do you guys want to further investigate this (I have daily database backups and could probably check out when this value changed with a bit of time) or can I close this issue? I´ll monitor it for a few further days anyway.

**Comment by bcutter** [comment]
I have no idea why but today I discovered the same error for the "value" column in table "nc_cards_properties".
- IS: 191
- SHOULD: 255

Changed it back to default (255), error/log entries disappeared. I suppose it´s been caused by a recent update, maybe one of the earlier majore ones failing in the first place and being performed manually afterwards.

Anyway, it seems to be a "for me only issue" therefore I´m closing it now.
